File: src/UpdateRules/glue_update.m
function rate = glue_update(s, m)
% Disassembly rate update rule
% Calculates rb from glue strength s and memory strength m
    % m>0 case
    rpos = (1-s)*(2/(1+exp(-m)) - 1);
    % m<0 case
    rneg = (s) * (2/(1+exp(-m)) - 1);
    rate = s + (m>0 * rpos) + (m<0 * rneg);
end

File: ystam/update_rules/glue_update.py
"""Glue update rule: ECM-modulated retention probability."""

from __future__ import annotations

import numpy as np


def glue_update(strength: float, memory: float) -> float:
    """Compute retention probability given glue strength and ECM memory.

    Args:
        strength: Glue binding strength in [0, 1]. Higher = more stable.
        memory: ECM signal at tile position.
            Positive memory stabilizes (increases retention).
            Negative memory destabilizes (decreases retention).

    Returns:
        Retention probability in [0, 1].
    """
    sigmoid = 2.0 / (1.0 + np.exp(-memory)) - 1.0
    r_pos = (1.0 - strength) * sigmoid
    r_neg = strength * sigmoid
    rate = strength + (memory > 0) * r_pos + (memory < 0) * r_neg
    return float(rate)
